lib/ueberauth.ex

```diff
@@ -344,8 +344,9 @@ defmodule Ueberauth do
   end
 
   defp run(conn, {module, :run_request, options}) do
-    to_request_path = Path.join(["/", conn.script_name, options.request_path])
-    to_callback_path = Path.join(["/", conn.script_name, options.callback_path])
+    route_prefix     = Path.join(["/" | conn.script_name])
+    to_request_path  = Path.join(["/", route_prefix, options.request_path])
+    to_callback_path = Path.join(["/", route_prefix, options.callback_path])
```

Comment:
```suggestion
    to_request_path  = Path.join([route_prefix, options.request_path])
    to_callback_path = Path.join([route_prefix, options.callback_path])
```
If `conn.script_name` is an empty list the previous method would return something like this `/./request_path`.